#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From b9b3045a036e869768cc31957156ed0d4c9840df Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Thu, 13 Jul 2023 14:23:44 +0200
Subject: [PATCH] Hack in some aboutdata for GnuPG Edition

This is just a quick hack to clarify that this is not
the upstream okular. No translations or anything.
---
 aboutdata.h | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/aboutdata.h b/aboutdata.h
index 90cfa9532..470efa8dc 100644
--- a/aboutdata.h
+++ b/aboutdata.h
@@ -16,7 +16,7 @@
 inline KAboutData okularAboutData()
 {
     KAboutData about(QStringLiteral("okular"),
-                     i18n("Okular"),
+                     i18n("Okular - GnuPG Edition"),
                      QStringLiteral(OKULAR_VERSION_STRING),
                      i18n("Okular, a universal document viewer"),
                      KAboutLicense::GPL,
@@ -28,6 +28,20 @@ inline KAboutData okularAboutData()
                      QString(),
                      QStringLiteral("https://okular.kde.org"));

+    about.setShortDescription(i18n("Okular, a lightweight PDF viewer with GnuPG integration."));
+#ifdef Q_OS_WIN
+    const QString othersource = QStringLiteral("from the <a href=\"https://apps.microsoft.com/store/detail/okular/9N41MSQ1WNM8\">Microsoft Store</a>");
+#else
+    const QString othersource = QStringLiteral("from your distribution.");
+
+#endif
+    about.setOtherText(QStringLiteral("Okular in general is a lightweight and highly secure document viewer for many"
+                        "document formats.<br/><br/>"
+                        "To reduce the attack surface even further the GnuPG Edition is stripped "
+                        "down to only support PDF documents without any active content.<br/>"
+                        "For the best User Experience you can safely install the fully featured Okular "
+                        ) + othersource);
+    about.setBugAddress("https://dev.gnupg.org/u/rgpg4win");
     about.addAuthor(i18n("Pino Toscano"), i18n("Former maintainer"), QStringLiteral("pino@kde.org"));
     about.addAuthor(i18n("Tobias Koenig"), i18n("Lots of framework work, FictionBook backend and former ODT backend"), QStringLiteral("tokoe@kde.org"));
     about.addAuthor(i18n("Albert Astals Cid"), i18n("Developer"), QStringLiteral("aacid@kde.org"));
--
2.41.0
